### DISCUSSION TITLE
A printing story-river as part of the core plugins?

### DISCUSSION THREAD

**Discussion by BurningTreeC** [TiddlyWiki/TiddlyWiki5#5757] [Ideas]
Hi there,

There has been request lately [here](https://groups.google.com/g/tiddlywiki/c/XNIL4l94mlQ) and [here](https://groups.google.com/g/tiddlywiki/c/txKKn-pTIfY) for a printing solution for TiddlyWiki5.
Some time ago I made a quick prototype of a printing plugin: printriver.tiddlyspot.com

My question: Should I create something similar as a tiddlywiki core plugin or a standalone plugin of my own?

**Comment by Jermolene** [comment]
My immediate thought is that this is plugin territory because the range of options that end users might plausibly want seems like it might be quite extensive.

But perhaps the core could incorporate some of the basics of the implementation, and provide tag-based hooks for plugins to extend things.

**Reply by BurningTreeC** [reply]
Thanks for your quick reply @Jermolene 
Which would be the basics of the implementation the core should have?

**Reply by Jermolene** [reply]
I was thinking that the core might include the print tiddler button in the sidebar and in the view template toolbar. The core functionality could just be to open the new window and include a "print" button. That could be done by rendering tiddlers tagged $:/tags/Print/Button, so that things could be extended.

**Comment by pmario** [comment]
I think the functionality the existing plugin has, would be a good starting point.
